src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -31,6 +31,7 @@ public override Task<Compilation> TransformCompilationAsync(Compilation oldCompi
                 }
 
                 public DocumentId DocumentId => _newState.Attributes.Id;
+                public override bool CanApplyToCompilationWithGeneratedDocuments => true;
             }
 
             internal sealed class TouchAdditionalDocumentAction : CompilationAndGeneratorDriverTranslationAction
```

Comment:
Ah, hmm, interesting. You're right this would be a good improvement for a additional file edit only. I think this is convincing me to make CanApplyToCompilationWithGeneratedDocuments default to true or make it abstract so it forces the thinking more carefully.

Reply:
>  CanApplyToCompilationWithGeneratedDocuments default to true \

plz no.

Reply:
> or make it abstract so it forces the thinking more carefully.

plz yes.

Reply:
It's now made abstract.